File: CS/WindowsApplication1/Form1.cs
using System;
using System.Windows.Forms;
using System.IO;
using System.Data;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) });
            return tbl;
        }
        

        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = CreateTable(20);
        }


        private void simpleButton1_Click(object sender, EventArgs e)
        {
            FileStream fs = new FileStream("test.xml", FileMode.Create);
            GridLayoutSerializer.SaveLayout(gridView1, fs);
            fs.Close(); 
        }

        private void simpleButton2_Click(object sender, EventArgs e)
        {
            gridView1.RestoreLayoutFromXml("test.xml");
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports System
Imports System.Windows.Forms
Imports System.IO
Imports System.Data

Namespace WindowsFormsApplication1

    Public Partial Class Form1
        Inherits Form

        Private Function CreateTable(ByVal RowCount As Integer) As DataTable
            Dim tbl As DataTable = New DataTable()
            tbl.Columns.Add("Name", GetType(String))
            tbl.Columns.Add("ID", GetType(Integer))
            tbl.Columns.Add("Number", GetType(Integer))
            tbl.Columns.Add("Date", GetType(Date))
            For i As Integer = 0 To RowCount - 1
                tbl.Rows.Add(New Object() {String.Format("Name{0}", i), i, 3 - i, Date.Now.AddDays(i)})
            Next

            Return tbl
        End Function

        Public Sub New()
            InitializeComponent()
            gridControl1.DataSource = CreateTable(20)
        End Sub

        Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim fs As FileStream = New FileStream("test.xml", FileMode.Create)
            GridLayoutSerializer.SaveLayout(gridView1, fs)
            fs.Close()
        End Sub

        Private Sub simpleButton2_Click(ByVal sender As Object, ByVal e As EventArgs)
            gridView1.RestoreLayoutFromXml("test.xml")
        End Sub
    End Class
End Namespace
